clients/git/e2e_test.go

```diff
@@ -19,55 +19,56 @@ import (
 	. "github.com/onsi/gomega"
 
 	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/clients/localdir"
 )
 
 var _ = DescribeTable("Test ListCommits commit-depth for HEAD",
 	func(uri string) {
 		const commitSHA = clients.HeadSHA
 		const commitDepth = 1
 		client := &Client{}
-		Expect(client.InitRepo(uri, commitSHA, commitDepth)).To(BeNil())
+		repo, err := localdir.MakeLocalDirRepo(uri)
+		Expect(err).To(BeNil())
+		Expect(client.InitRepo(repo, commitSHA, commitDepth)).To(BeNil())
 		commits, err := client.ListCommits()
 		Expect(err).To(BeNil())
 		Expect(len(commits)).Should(BeEquivalentTo(commitDepth))
 		Expect(client.Close()).To(BeNil())
 	},
-	Entry("GitHub", "https://github.com/ossf/scorecard"),
-	Entry("Local", "file://../../"),
-	Entry("GitLab", "https://gitlab.haskell.org/haskell/filepath"),
```

Comment:
Was there a reason to get rid of the non-local URIs (here and elsewhere)? The client does a git clone for them.

Reply:
I assumed for the tests when we use `localdir.MakeLocalDirRepo` it can't handle other than local repo.

Also why would want to test with a GitHub repo for local git repo?

Reply:
> Also why would want to test with a GitHub repo for local git repo?

The git client also does a git clone from remotes, so testing with GitHub/GitLab tests that portion

Reply:
The ` localdir.MakeLocalDirRepo(uri)` doesn't handle GitHub or GitLab.

Reply:
Yes, but the test harness doesn't need to assume a local dir, right? Can we use the first part of the entry (`Local`, `GitHub`, `GitLab`) to create the correct `clients.Repo` implementation?

Reply:
There are issues with the GitLab client parsing the URL. 

I tried debugging it, which requires much more effort to resolve. 

Can we do this as a subsequent PR?

Reply:
Is it just due to your lines here?
```go
func getRepoClient(uri string) (clients.Repo, error) {
	if strings.Contains(uri, "github.com") {
		return githubrepo.MakeGithubRepo(uri)
	} else if strings.Contains(uri, "gitlab.com") {
		return gitlabrepo.MakeGitlabRepo(uri)
	}
	return localdir.MakeLocalDirRepo(uri)
}
```

But the test repo file is: `https://gitlab.haskell.org/haskell/filepath` which has `gitlab` in it, but not `gitlab.com` ?

Reply:
Thanks @spencerschrock